qpid/cpp/src/qpid/broker/amqp/Outgoing.cpp

```diff
@@ -79,6 +79,9 @@ OutgoingFromQueue::OutgoingFromQueue(Broker& broker, const std::string& source,
 
 void OutgoingFromQueue::init()
 {
+    // observe queue for changes to track undeliverable messages
+    queue->getObservers().add(boost::shared_ptr<QueueObserver>(shared_from_this()));
+
```

Comment:
If possible, I'd really like to make the tracking lazy, only activating if actually requested/needed, and only then registering the queue observer and tracking the messages.